### ISSUE TITLE
[Project] EEG/Biosignal Classifier App 

### ISSUE THREAD

**Issue by mori-c** [gdg-toronto/firebase-study-jam-season-1#3]
![](https://media1.giphy.com/media/QC9MRxmLskle8/giphy.gif)

## Idea

Brain-Computer Interface classification feature could identify lobe activity strengths and channel output averages per node using FFT algorithm. 

If time permits within study group schedule, add sentimental analysis as a soft reinforcement learning output

## Goal

Create an EEG Biosignal classifier with PyTorch/Tensorflow connected with Firebase

## TODO

- [ ] find out how to connect pytorch/tensorflow models into ios

## Process

* jupyter notebook (code)
* firestone, firebase (database)
* pytorch (library simulator)
* flask (server requests)
* django (frontend)
* muse api (ios)

Backup:

> in case pytorch models don't integrate well

* colab notebook (code)
* firestone, firebase (database, server requests)
* tensorflow (library simulator)
* angularJS (frontend)
* muse api (ios)

## Breakdown

| ✅ | Process  | I / O  | Method / Tools       | 
| --- | ------------------------- | ------ | ---------------------- |
|  | store/upload data | input  | firestone      			 	|
|  | clean eeg data  | input  | colab, numpy, pandas       	|
|  | graph, visualize data   | output | colab, matplotlib      |
|  | identify frequency channels         | ...      | ...          |
|  | remove interfernce noise        | input  | firestone, csv |
|  | apply deep learning algorithms (FFT)  | input  | colab, algebra  |
|  | display output of either image or wavelengths (tbd)  | output | ... |
|  | identify loss perhaps around less active channels by boosting (reinforce activity with auditory engagement, for example) | ...      | colab, generalization, loss, monte carlo / bayesian / regression |
|  | > to be continue | ... | ... |

## Firestone (database) → Firebase Setup

[Firestore Sub-Collections](https://howtofirebase.com/firestore-sub-collections-2e23c998540d)

## Framework

[Django](https://www.hackanons.com/2018/03/python-django-with-google-firebase.html) or [Angular](https://angular.io/)

## Repo

Documentation and code found under [mori-c/meetups/gdg](https://github.com/mori-c/meetups/tree/master/gdg)

## Notes

Based on [neurotechnology](https://github.com/mori-c/neurotech) coursework completed recently will inspire most of the methods used to execute this webapp
